libalpm/scripts/mkinitcpio-install

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash -e
+
+args=()
+all=0
+
+while read -r line; do
+    if [[ $line != */vmlinuz ]]; then
+        all=1
+        continue
+    fi
+
+    pkgbase=$(<"${line%/vmlinuz}/pkgbase")
```

Comment:
mapfile is still fine. `read` would also be fine. Either one is fewer subprocesses.

Reply:
Sure, I'll change the implementation to use either mapfile or read.

Reply:
Use `read -r pkgbase < "${line%/vmlinuz}/pkgbase"`.

`mapfile` is reading a set of lines into an array.

Reply:
The first revision read a set of lines into an array, using "append to end of array" mode:

```
mapfile -O ${#targets[@]} -t targets < "/$line"
```

Yes, if you want a non-array variable (the script has changed a lot since the initial one I hacked up) then read is your friend.

Reply:
Implemented on 9c1e0f3dd01151b79e9b557ea8510605616c1369